src/pathfinder.rs

```diff
@@ -0,0 +1,367 @@
+        RegisteringNodesError::SerdeError(error)
+    }
+}
+
+impl From<mut_static::Error> for RegisteringNodesError {
+    fn from(error: mut_static::Error) -> Self {
+        RegisteringNodesError::MutexError(error)
+    }
+}
+
+impl std::fmt::Display for RegisteringNodesError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            RegisteringNodesError::MutexError(e) => write!(f, "Mutex error : {e}"),
+            RegisteringNodesError::SerdeError(e) => write!(f, "Parsing error : {e}"),
+            RegisteringNodesError::NodesNotCorrectlyIndexed => {
+                write!(f, "Node not indexed properly")
+            }
+        }
+    }
+}
+
+byond_fn!(fn register_nodes_astar(json) {
+    if let Err(e) = NODES.set(Vec::new()) {
+        return Some(format!("{e}"));
+    }
+    let mut nodes = NODES.write().unwrap();
+    match register_nodes(json, nodes.as_mut()) {
+        Ok(s) => Some(s),
+        Err(e) => Some(format!("{e}"))
+    }
+});
+
+// Builds a list of nodes from a json file.
+// Errors if the input list of nodes is not correctly indexed. Each node should have for unique id its position in the list, with the first unique-id being 0.
+fn register_nodes(
+    json: &str,
+    nodes: &mut Vec<NodeContainer>,
+) -> Result<String, RegisteringNodesError> {
+    let deserialized_nodes: Vec<Node> = serde_json::from_str(json)?;
+    if deserialized_nodes
+        .iter()
+        .enumerate()
+        .filter(|(i, node)| i != &node.unique_id)
+        .count()
+        != 0
+    {
+        return Err(RegisteringNodesError::NodesNotCorrectlyIndexed);
+    }
+
+    deserialized_nodes
+        .into_iter()
+        .for_each(|node| nodes.push(NodeContainer::new(node)));
+
+    Ok("1".to_string())
+}
+
+byond_fn!(fn add_node_astar(json) {
+    let mut nodes = NODES.write().unwrap();
+    match add_node(json, nodes.as_mut()) {
+        Ok(s) => Some(s),
+        Err(e) => Some(format!("{e}"))
+    }
+});
+
+// Add a node to the static list of node.
+// If it is connected to other existing nodes, it will update their connected_nodes_id list.
+fn add_node(json: &str, nodes: &mut Vec<NodeContainer>) -> Result<String, RegisteringNodesError> {
+    let new_node: Node = serde_json::from_str(json)?;
+
+    // As always, a node unique id should correspond to its index in NODES
+    if new_node.unique_id != nodes.len() {
+        return Err(RegisteringNodesError::NodesNotCorrectlyIndexed);
+    }
+
+    // Make sure every connection we have we other nodes is 2 ways
+    new_node.connected_nodes_id.iter().for_each(|index| {
+        if let Some(node_container) = nodes.get_mut(*index) {
+            if let Some(node) = node_container.node.write().unwrap().as_mut() {
+                node.connected_nodes_id.push(new_node.unique_id)
+            }
+        };
+    });
+
+    nodes.push(NodeContainer::new(new_node));
+
+    Ok("1".to_string())
+}
+
+enum DeleteNodeError {
```

Comment:
Remove all your custom stuff and just use [thiserror](https://docs.rs/thiserror/latest/thiserror/) (which we already include).